### DISCUSSION TITLE
FastAPi integration - RSGI

### DISCUSSION THREAD

**Discussion by KaliszS** [emmett-framework/granian#126] [Q&A] [answered]
Hello. I'm trying to replace uvicorn with granian in my API based on the FastAPI framework, but using the RSGI interface. However, I am encountering problems (`FastAPI.__call__() missing 1 required positional argument: 'send'`). Do you know the easiest way to run such a configuration?

**Comment by gi0baro** [comment, marked as answer]
Starlette and thus FastAPI don't support RSGI, you can use Granian with the ASGI interface using the relevant parameter

**Comment by sreenivasulureddysura** [comment]
@gi0baro Is there any feature support for RSGI?

**Reply by mcharytoniuk** [reply]
That should be the question to FastAPI. :)

**Reply by gi0baro** [reply]
Actually to Starlette, as FastAPI doesn't manage ASGI by itself.
And it was already asked: https://github.com/encode/starlette/discussions/2287
